search/utils.py

```diff
@@ -82,6 +82,7 @@ def doc_type_warning(func):
     @functools.wraps(func)
     def wrapper(self_or_cls, *args, **kwargs):
         if kwargs.get("doc_type"):
+            kwargs.pop("doc_type")
```

Comment:
please add default value to the pop method, otherwise, KyeError will raise in case key not found.

Reply:
we check if doc_type exists in kwargs before pop

Reply:
it looks a little bit weird, I would use a little bit different approach 
```
doc_type = kwargs.pop("doc_type", None)
if doc_type is not None:
```
But it up to you

Reply:
Done, thanks